tck/src/main/java/ee/jakarta/tck/concurrent/spec/Platform/virtual/VirtualThreadServlet.java

```diff
@@ -0,0 +1,468 @@
+
+        future = platformManagedScheduledExecutorDD.submit(Thread::currentThread);
+        assertFalse(isVirtual(future.get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS)));
+    }
+
+    public void testVirtualScheduledExecutor() throws Exception {
+        ManagedScheduledExecutorService virtualManagedScheduledExecutorAnno = InitialContext
+                .doLookup("java:app/concurrent/ManagedScheduledExecutorAnnoVirtual");
+        ManagedScheduledExecutorService virtualManagedScheduledExecutorDD = InitialContext
+                .doLookup("java:app/concurrent/ManagedScheduledExecutorDDVirtual");
+
+        assertNotNull(virtualManagedScheduledExecutorAnno);
+        assertNotNull(virtualManagedScheduledExecutorDD);
+        
+        Thread annoThread = virtualManagedScheduledExecutorAnno.submit(Thread::currentThread)
+                .get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS);
+        Thread ddThread = virtualManagedScheduledExecutorDD.submit(Thread::currentThread)
+                .get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS);
+        
+        if (VERSION == 17) { //TODO remove when Concurrency API supports only 21+
+            assertThrows(NoSuchMethodException.class, () -> isVirtual(annoThread), "Should be impossible to get a virtual thread on Java 17");
+            assertThrows(NoSuchMethodException.class, () -> isVirtual(ddThread), "Should be impossible to get a virtual thread on Java 17");
+            return;
+        }
+
+        // Java 21+
+        assertEquals(isVirtual(annoThread), isVirtual(ddThread));
+        assumingThat(isVirtual(annoThread) && isVirtual(ddThread), () -> {
+            
+            // Test schedule on virtual threads
+            final LinkedBlockingQueue<Object> results = new LinkedBlockingQueue<>();
+            ScheduledFuture<?> oneTimeFuture = virtualManagedScheduledExecutorDD.schedule(
+                    (Runnable) new LookupActionCaptureThread(results,
+                            "java:app/concurrent/ManagedScheduledExecutorDDVirtual"),
+                    TestConstants.pollInterval.toMillis(), TimeUnit.MILLISECONDS);
+    
+            assertTimeoutPreemptively(TestConstants.waitTimeout, () -> {
+                for (; 2 != results.size(); Wait.sleep(TestConstants.pollInterval)) {
+                    // Wait till we have exactly 2 results indicating 1 run of schedule
+                }
+            });
+    
+            assertTrue(oneTimeFuture.isDone());
+            assertEquals(2, results.size());
+    
+            Thread thread = (Thread) results.poll(1, TimeUnit.MILLISECONDS);
+            Object result = results.poll(1, TimeUnit.MILLISECONDS);
+    
+            if (result instanceof Throwable)
+                throw new AssertionError("An error occured on thread.", (Throwable) result);
+            assertTrue(isVirtual((Thread) thread));
```

Comment:
Similar to first comment. This is not a requirement, but very unlikely it would not be the case.